prometheus/histogram.go

```diff
@@ -1835,3 +1835,131 @@ func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
 		n.exemplars = append(n.exemplars[:nIdx], append([]*dto.Exemplar{e}, append(n.exemplars[nIdx:rIdx], n.exemplars[rIdx+1:]...)...)...)
 	}
 }
+
+type constNativeHistogram struct {
+	desc                            *Desc
+	count                           uint64
+	sum                             float64
+	labelPairs                      []*dto.LabelPair
+	nativeHistogramSchema           int32
+	nativeHistogramZeroThreshold    float64
+	nativeHistogramMaxZeroThreshold float64
+	createdTimestamp                time.Time
+	nativeExemplars                 []*dto.Exemplar
+
+	positiveBuckets map[int]int64
+	negativeBuckets map[int]int64
+	zeroBucket      uint64
+}
+
+func NewConstNativeHistogram(desc *Desc, count uint64, sum float64, postiveBuckets, negativeBuckets map[int]int64, zeroBucket uint64,
```

Comment:
Needs doc string to explain the arguments and the fact that this is exposed as a counter type histogram, treated as a counter on scrape.

Reply:
Tried this.  Wouldn't this blow up the argument list? Is it better to create a seperate a constNaticveGuageHistogram ?

Reply:
I'm still unclear about including a ` gauge histogram` . As per my understanding , we will need to add  postiveBuckets, negativeBuckets of type  map[int]float64 as an argument (or use a generic parameter).

Reply:
It sounds like you are thinking about a float histogram. That's something we might want to add later (ideally together with the option of creating classic float histograms). However, the use case for float histograms in instrumentation is very rare.

I am in fact talking about a gauge histogram. It merely requires to set a different `MetricType` in the `MetricFamily`. However, I've just realized that we currently have no way of communicating the difference of counter histogram (i.e. the "normal" histogram) and gauge histogram during metrics collection. For that, we needed `Desc` to have a metric type field (which was the plan for quite some time).

So I'm pretty sure we cannot solve this problem in this PR, because it needed too much re-architecting.

But if we assume that the difference between counter histogram and gauge histogram will be in the `Desc`, we actually don't need a bool parameter here, but just have to later allow the option of creating "typed" `Desc`s.

In conclusion, let's not do gauge histograms right now.

Reply:
That's true. I was indeed confused about a float histogram and gauge histogram.